gc/structs/ForwardedHeader.hpp

```diff
@@ -497,6 +497,24 @@ class MM_ForwardedHeader
 	}
 #endif /* defined(OMR_GC_COMPRESSED_POINTERS) */
 
+	MMINLINE uint32_t
+	getObjectCompressedSize()
+	{
+		/* in compressed headers, the size of the object is stored in the low-order half of the uintptr_t read when we read clazz
+		 * so read it from there instead of the heap (since the heap copy would have been over-written by the forwarding
+		 * pointer if another thread copied the object underneath us). In non-compressed, this field should still be readable
+		 * out of the heap.
+		 */
+		uint32_t size = 0;
+#if defined (OMR_GC_COMPRESSED_POINTERS)
+		if (compressObjectReferences()) {
+			size = getPreservedOverlap();
+		}
+#endif /* defined (OMR_GC_COMPRESSED_POINTERS) */
+
+		return size;
+	}
+
```

Comment:
So this is a part of the initiative to move  #ifdef COMPRESSED inside Overlap group of methods, to simplify the call sites?

Not clear how you planned to use this method, but regardless, not very enthusiastic about it.

For the most part I think you can just remove that #ifdef from that group of methods (but I might be oversimplifying it!?). 

Some of those methods already have 

ForwardedHeaderAssert(compressObjectReferences());

at the front to make sure callers don't misuse it, but we might need to add it to the rest of them, too.

Reply:
Btw, ForwardedHeaderAssert is not enabled by default (would be too expensive for production builds), so whatever you end up doing a round of testing with ForwardedHeaderAssert enabled (have to recompile whole GC) is recommended.

Reply:
> So this is a part of the initiative to move #ifdef COMPRESSED inside Overlap group of methods, to simplify the call sites?

That's correct. I'm also not that fond of this either. 

> For the most part I think you can just remove that #ifdef from that group of methods (but I might be oversimplifying it!?).

That could work

Reply:
The removal of `#ifdef` worked for our purpose

Reply:
But it looks like for OMR compilation (32bit system) we need these macros in place after all. Since we cannot remove these `#ifdef`s I would say to keep things the way they are (even at call sites)?